ServerlessLibraryUI/src/components/AppItem.vue

```diff
@@ -69,14 +69,32 @@ export default {
     }
   },
   methods:{
-     showConsentModal(data) {
-        this.isModalVisible = true;
-        this.modalData = data;
-      },
+    showConsentModal(item) {
+      this.isModalVisible = true;
+      this.modalData = item;
+    },
+
+    outboundRepoClick(item) {
+      this.trackEvent('/sample/source', item);
+    },
 
+    trackEvent(eventName, item) {
+      if (typeof appInsights !== 'undefined') {
+        const eventData = this.getDataToLog(item);
+        appInsights.trackEvent(eventName, eventData);
+      }
+    },
+
+    getDataToLog(item) {
+      return {
+        repository: item.repository,
+        template: item.template
+      };
+    }
   },
   filters: {    
     ToDisplayLanguage(value) {
```

Comment:
I dont recollect how these filters get called. I dont see any with a `value` being passed . Here is the only reference I could find.  
https://github.com/Azure/ServerlessLibrary/blob/4579294ad0aa7109fa78b8e7865ce1a870bf5347/ServerlessLibraryUI/src/components/AppItem.vue#L20 
If so should we refactor this method ?

Reply:
From [https://vuejs.org/v2/guide/filters.html](url) :
Filters should be appended to the end of the JavaScript expression, denoted by the “pipe” symbol. 
The filter’s function always receives the expression’s value (the result of the former chain) as its first argument.